6_survival_analysis/nbconverted/survival_hit_groups.py

```diff
@@ -0,0 +1,704 @@
+                    label=get_label(gene, is_mutated, n_mutant, n_wildtype))
+
+        ax.legend()
+
+        def get_n_hits(hits_class, ix):
+            hits_desc = (['one', 'two'] if hits_class == 'one' else ['two', 'one'])
+            return hits_desc[ix]
+
+        ax.set_title('Survival for {}-hit classification'.format(
+            get_n_hits(hits_classification, ix)))
+
+        # hypothesis testing using log-rank test
+        y = Surv.from_dataframe('status', 'time_in_days', id_clinical_df)
+        chisq, p_val = compare_survival(y, id_clinical_df[mut_col].values)
+        print(mut_col, 'chisq = {:.4f}'.format(chisq), 'p = {:.4e}'.format(p_val))
+
+    plt.suptitle('Comparing mutation classes for {}'.format(identifier))
+    
+plot_id(identifier, id_clinical_df)
+
+
+# ### Run for all "class 2" genes
+# 
+# These are all tumor suppressors that Park et al. have annotated as "two-hit loss" drivers (i.e. classical tumor suppressors).
+# 
+# We want to see how many of them (if any) distinguish between survival groups more effectively for two-hit samples as opposed to one-hit samples. On average, if their classes make sense we would expect to see that a comparison between two-hit samples and all other samples gives us "better" survival groups than a comparison between 1+ hit samples and 0-hit samples.
+# 
+# We'll quantify "better" using a log-rank test with the above groups, and look at the difference in test statistics. This is going to be somewhat dependent on sample size but it should give us a general idea of whether or not there's any survival-related signal in these mutation groups across genes/cancer types.
+
+# In[21]:
+
+
+class_2_ids = park_loss_sig_df.index.unique()
+print(len(class_2_ids))
+print(class_2_ids[:10])
+
+
+# We want to build a dataframe with:
+# * identifier
+# * \# WT samples
+# * \# mutant samples
+# * survival p-value
+# * \# WT samples (alt classification)
+# * \# mutant samples (alt classification)
+# * survival p-value (alt classification)
+
+# In[22]:
+
+
+class_2_surv_df = []
+columns = [
+    'identifier',
+    'n_mutant',
+    'n_wildtype',
+    'chisq',
+    'p_val',
+    'n_mutant_alt',
+    'n_wildtype_alt',
+    'chisq_alt',
+    'p_val_alt',
+]
+
+for identifier in class_2_ids:
+    
+    results = [identifier]
+    
+    id_clinical_df = get_groups_for_gene_and_tissue(
+        identifier, 'TSG', 'two')
+    
+    for ix, mut_col in enumerate(['is_mutated', 'is_mutated_alt']):
+    
+        n_mutant = id_clinical_df[mut_col].sum()
+        n_wildtype = id_clinical_df.shape[0] - n_mutant
+        
+        if n_mutant == 0 or n_wildtype == 0:
+            print(identifier, mut_col, n_mutant, n_wildtype, file=sys.stderr)
+            continue
+    
+        # hypothesis testing using log-rank test
+        try:
+            y = Surv.from_dataframe('status', 'time_in_days', id_clinical_df)
+        except ValueError:
+            # this happens for COADREAD, TODO fix it later
```

Comment:
Again, just checking you didn't mean to do this prior to the PR. Your note is definitely fine though.